xchange-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/dto/PoloniexOrderbook.java

```diff
@@ -27,19 +27,24 @@ public void modify(OrderbookModifiedEvent modifiedEvent) {
         BigDecimal price = modifiedEvent.getPrice();
         BigDecimal volume = modifiedEvent.getVolume();
 
-        side.remove(price);
-        if (volume.compareTo(zero) != 0) {
-            side.put(price, volume);
+        synchronized(side) {
```

Comment:
Have you considered switching `OrderbookModifiedEvent` to use `ConcurrentSkipListMap` rather than `TreeMap`?